File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
# [1,5,3,5,1,2,1,4]
def solution(board, moves):
    answer = 0
    basket = []
    for i in moves:
        i -= 1
        for j in range(len(board)):
            if board[j][i] == 0:
                continue
            
            else:
                if (len(basket) == 0) or (len(basket) != 0 and basket[-1] != board[j][i]):
                    basket.append(board[j][i])
                    board[j][i] = 0
                    break
                    
                if len(basket) != 0 and basket[-1] == board[j][i]:
                    board[j][i] = 0
                    basket.pop()
                    answer += 2
                    break
    return answer

File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.ArrayList;

public class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        ArrayList<Integer> basket = new ArrayList<>();
        for (int i : moves) {
            i--;
            for (int j = 0; j < board.length; j++) {
                if (board[j][i] == 0) {
                    continue;
                } else {
                    if (basket.isEmpty() || basket.get(basket.size() - 1) != board[j][i]) {
                        basket.add(board[j][i]);
                        board[j][i] = 0;
                        break;
                    }
                    if (!basket.isEmpty() && basket.get(basket.size() - 1) == board[j][i]) {
                        board[j][i] = 0;
                        basket.remove(basket.size() - 1);
                        answer += 2;
                        break;
                    }
                }
            }
        }
        return answer;
    }
}
